### DISCUSSION TITLE
Curious on which is better?  Pre rewrite, or after rewrite?  Other comments on bugs?

### DISCUSSION THREAD

**Discussion by PerryCS** [azerothcore/azerothcore-wotlk#14053] [General]
EDIT: I did an update, see at bottom...

I'm using - AzerothCore rev. b7f80839151f 2021-11-10 23:43:00 +0100 (master branch) (Unix, RelWithDebInfo, Static)

First of all, I am not complaining - I love the fact I have a private server to scratch my wow itch.  I have had versions of AzerothCore (AC) before the rewrite 1 year ago approx and the one I am running now which is approx from Nov 2021 or a few versions after the re-write.  A HUGE thank you to everyone contributing.

My question is this... and it's sincere... not trolling or trying to upset anyone (I'm 53 years old - too tired to be a troll)... but, every so often I come to the repositories and I see pages of bugs for example, recently in The Underbog.  I am curious... how is the state of the game today vs say a few weeks after the re-write vs before the rewrite.

So far in my playing the older versions - one problem was the hunter pet wouldn't follow properly.  That works fine in AC.  That also works fine in the re-write version I am currently running.

Are there more quests and stuff bugged now because of the rewrite or are these issues (say Underbog) always been there and it's only now someone is flagging them?

I'm curious what caused "all of a sudden" all sorts of problems with Underbog.  I have never upgraded by server in the past 10 years of running my own private server.  I have always re-created it from scratch and re-leveled.  I was eager to upgrade recently but then saw all these pages and pages of problems with Underbog and thought to myself... did someone commit a change to make the code better and in doing so, broke a few things that now need to be fixed?  Is this change better overall, or worse?  Will all these new problems be fixed in a few months?  Should I wait to upgrade?

So, just curiousity.  I wish I had the time and skill to help with this project.  I can program 30+ languages including 6502 assembly (i'm old) LOL!... but, I find it hard to fix other peoples' code.  The learning curve is quite high it seems.  I keep reading the source code from time to time when I have a spare moment to see if it's something I can quickly grasp and help out around here but, so far, I'm lost in helping so I just rely on all of you.

With that being said, I am curious on the general well being I guess you could say of the project.  !?!??!  

Also, how did private servers run so many Cataclyism, MOP, etc servers?  I wish one of them would release their source as it seems quite a few hardcore private servers were so close to retail you could barely tell.   And then I look on here, and think, wow, so many bugs... I wish I had the knowledge to help fix them.   I had a Trinty Core version waaaay back but that was one of the versions the hunter pet in the starting area kinda screwed up.  AC has been close to perfect so far.  A few quests bugged.  A few clipping issues with the map, mobs walking through stuff, partially cut off in the floor... but, not enough to ruin the game-play.  I mostly just quest around by myself in the game when I have that nostalgia itch...

I'm just curious... what is the most retail like of them all?  With the least bugs?  I know pre Blizz offering retail Classic this project had piles of people daily fixing things... not sure what it's like now that Blizz real servers are available.

Any comments?

Thank you for your time :)

All I want, is as close to retail as possible.  Would upgrading what I have now (approx 1 year old version or whenever the re-write was) vs now?

******************************************************************************************
UPDATE: I was thinking to myself... wait!    I have this running in a VM... why don't I TRY to update this and SEE how good it is.  If it doesn't work (happened 4+ years ago I could never update it) then I just restore the VM's snapshot.

SO... I updated it... I am shocked at how much easier it is to UPGRADE the server now...

.server debug
then go the full upgrade... you even had the intermediate update there for me...
git pull; git checkout 2d609e9e48b2e516e4a555015b2662cab15b0c38
then I remembered running the acore to update the DB....
THAT worked flawlessly.  
THEN... I recompiled. Took about 15 min.  (Ryzen 2700x 32gb of ram) but the drive is on an array not an SSD.
I ran the authserver - it updated a few things.  DONE!  Smooth!
I ran the worldserver - whoever did the UPDATE part... did it perfect... it started back in 11-2021 and did perfect DB updates right up to present.
I ran the worldserver... gave me an error on the MAPS...
I don't know how to do it the simple way without sucking some bandwidth from github but I just downloaded v15 latest off the website.  Extracted into it's old folder.   Deleted everything in the azeroth-server data folder and cut and pasted my extracted files in there.

Ran it.   INSTANTLY worked!
ALSO... before my upgrade... I was standing beside Nat Pagle and once completing a quest he had a "Vender" symbol when I clicked on him (a BUG).   The second I loaded up, went into my character... POOF... it was the proper icon!  Even that was fixed!

So, SO FAR!   Seconds into the new build as of today... I already noticed a fix!   Also, the sheer number of fixes that scrolled by... THANK YOU all for making the update easy for this old guy!  I may know assembly language but that was before most of you were born prob!  LOL!   6502 is a Commodore 64/Vic 20 assembly I learned on.  LOL!

One day I hope to help this project but I'm getting older... less time in a day... not retiring for 13+ years... so, we shall see...

I'll play on it and see what's up.   Is there a place I can "generalized info" on the status of the project?  Like plans or projections?  For example... maybe a few months ago someone wanted to re-write something... a change log would be cool but would prob add more work.   Just would be nice to see more than the 1.3k bugs listed and then be wondering... is it better?  is it worse?  Who knows?  The devs do!   I don't...   

So far, I like the change.  1 bug fixed right before my eyes :)

**Comment by PerryCS** [comment]
I have been playing the updated AC from 3 weeks ago.  There seems to be a lot more bugs in the game so I am assuming there was a recent re-write of something to trigger the flood of problems I see in the main page.  Now close to 1.4k - I wish I had time to help squash some of these bugs.

I am curious which version of the AC is the one before all these changes?  Maybe I could rollback to that before all this stuff became broken?  I know sometimes you have to break some eggs to move forward - not complaining.  But, I would like a more realistic Blizzard experience.

I switched from TC to AC a few years ago and I found AC to be the closer experience (pets actually followed you properly in AC but not in the TC version I had installed).

I'm just wondering if anyone could explain the sudden explosion of problems?  Was there an update that broke a bunch of stuff?  I know eventually all this stuff will be fixed, and as I said, I understand that sometimes to move forward smoothly you need to break a few things.

Just curious basically what happened and if anyone knows the version that was before all these changes?

**Comment by Nyeriah** [comment]
Hello @PerryCS 

Azerothcore's main goal is to provide a completely bug free experience and to achieve that we're constantly refining and revisiting old content.

To support such goal, we have the Chromiecraft project's help (https://www.chromiecraft.com/en/), progressing through every stage of the game (vanilla, tbc and in the future wotlk), while we try to address every issue we are able to identify.

The sudden increase of issues you've noticed is thanks to the fact the Chromiecraft project has moved to the Burning Crusade expansion, and thus those issues that were present there are now reported. They aren't new issues, they just were not indentified yet.

Thanks for your feedback and contributions,

Regards.

**Reply by PerryCS** [reply]
OH!   That's why I have been seeing more ChromieCraft.   I have also noticed more "divisions" like... Content 1-10 Content 11-20 type of tags.   THANK YOU for explaining it to me.  So, I guess those issues were always there then, just now being revealed.

I was thinking, they made a HUGE change which broke a lot of stuff in order to move ahead easier.   Good to know I'm running a decent version.  My AC is from 3 weeks ago.  I'll update again when I have some time.

I'll check out ChromieCraft for more info.   Thanks :)

**Reply by PerryCS** [reply]
I just checked out ChromieCraft!  YES!   That is the info I was looking for!   I see what they are trying to do!   Awesome!   Thanks for pointing me in the right direction.  :)

I understand now why things are the way they are.  I was confused before seeing all the errors and figured... they broke something on purpose to re-flow the code, or fix a core problem, etc...

Happy to know what ChromieCraft is after seeing it for quite a while :)

**Reply by PerryCS** [reply]
YES!   They talk about a concern I have had for years!   Why is it that private servers fix all sorts of bugs, based on the open source foundation, and then when they go down, they take the fixes with them.  I always wondered why they wouldn't allow the source to be revealed so others could benefit.

EXCELLENT!  This makes me so happy!  :)
